### PR TITLE
feat: implement browser testing

### LINKED ISSUE #284: Create test to `browser compatibility`
`Fuels-ts` would probably be used on many WebApps and this requires adding tests to run jest on a browser environment.

See also:
- https://github.com/FuelLabs/fuels-ts/issues/651
- https://github.com/FuelLabs/fuels-ts/issues/649
- https://github.com/FuelLabs/fuels-ts/issues/309
- https://github.com/FuelLabs/fuels-ts/issues/264

### LINKED ISSUE #765: Add `Buffer` compatibility for browsers
We need to support `Buffer` usage inside the Browsers.

For example, someone using React with Vite might fall into this error:

error

And then having to resort to custom mapping configurations such by installing the `buffer` package:

```console
pnpm install buffer
```

And then configuring custom mappings in `vite.config.js`:

```ts
import { defineConfig } from 'vite'
import react from '@vitejs/plugin-react'

// https://vitejs.dev/config/
export default defineConfig({
  plugins: [react()],
  resolve: { alias: { buffer: 'node_modules/buffer/index.d.ts' } },
})
```

### PR DESCRIPTION
**Pull request by danielbate**
- Closes #284
 - Closes #765

### CODE REVIEW

File: packages/versions/src/lib/getSystemVersions.test.ts [hunk 1]
```diff
@@ -2,12 +2,18 @@ import * as childProcessMod from 'child_process';
 
 import { getSystemVersions } from './getSystemVersions';
 
-// https://stackoverflow.com/a/72885576
-jest.mock('child_process', () => ({
-  __esModule: true,
-  ...jest.requireActual('child_process'),
-}));
+vi.mock('child_process', async () => {
```
Comment (petertonysmith94) [inline packages/versions/src/lib/getSystemVersions.test.ts:5]: I personally believe the following format is more readable

```ts
vi.mock('child_process', async () => ({
  __esModule: true,
  ...await vi. importActual('child_process'),
}));
```

File: scripts/tests-browser.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
+import { execSync } from 'child_process';
+import { readFileSync, readdirSync, writeFileSync } from 'fs';
+import { join } from 'path';
+
+const initialDependencies: { file: string; contents: string }[] = [];
+
+/**
+ * - Needs refactoring
```
Comment (petertonysmith94) [inline scripts/tests-browser.ts:8]: Todo

File: packages/providers/test/provider.test.ts [hunk 3]
```diff
@@ -253,9 +259,12 @@ describe('Provider', () => {
...
      * to ensure it'll be properly overriden in `connect` method below
      */
-    const fetchChainAndNodeInfo = jest
+    const fetchChainAndNodeInfo = vi
       .spyOn(Provider.prototype, 'fetchChainAndNodeInfo')
-      .mockImplementation();
+      .mockResolvedValue({
+        chain: {} as ChainInfo,
+        nodeInfo: {} as NodeInfo,
+      });
```
Comment (petertonysmith94) [inline packages/providers/test/provider.test.ts:267]: @danielbate why has this changees out of curiosity?

**Comment by danielbate** [comment]
Superceded by #1630

**Pull request closed without merging**